tests/plugin_http_api_test.py

```diff
@@ -58,6 +90,9 @@ def startEnv(self) :
         nodeos_flags = (" --data-dir=%s --config-dir=%s --trace-dir=%s --trace-no-abis --access-control-allow-origin=%s "
                         "--contracts-console --http-validate-host=%s --verbose-http-errors --max-transaction-time -1 --abi-serializer-max-time-ms 30000 --http-max-response-time-ms 30000 "
                         "--p2p-peer-address localhost:9011 --resource-monitor-not-shutdown-on-threshold-exceeded ") % (self.data_dir, self.config_dir, self.data_dir, "\'*\'", "false")
+        nodeos_flags += category_config.nodeosArgs()
+
+        print("nodeos_flags = ", nodeos_flags)
```

Comment:
Remove debug print or convert to Utils.Print.

Reply:
removed

Reply:
> I'm not understanding where we do the `open()/bind()/listen()` on the acceptor in this new version?
I guess this is what you don't get.
https://www.boost.org/doc/libs/1_82_0/doc/html/boost_asio/reference/basic_socket_acceptor/basic_socket_acceptor/overload5.html

Reply:
> > I'm not understanding where we do the `open()/bind()/listen()` on the acceptor in this new version?
> > I guess this is what you don't get.
> > https://www.boost.org/doc/libs/1_82_0/doc/html/boost_asio/reference/basic_socket_acceptor/basic_socket_acceptor/overload5.html

Yes, that's it, thanks. This doesn't appear to do the error checking and throws we used to do. Could it be an issue?

Reply:
All asio apis throw std::system_error when there's no ec input parameter.

Reply:
Yes, but before we called:
```
void fail(beast::error_code ec, char const* what, fc::logger& logger, char const* action) {
   fc_elog(logger, "${w}: ${m}", ("w", what)("m", ec.message()));
   fc_elog(logger, action);
}
```
which probably logged a more descriptive message describing the issue.

Reply:
In general allowing the framework to do more of the work seems like a win. We do need to be careful about exceptions, see https://github.com/AntelopeIO/leap/pull/1175 for example. However, this is only called during startup, so seems safe to allow it to throw and report the error.

However, we do need `http_plugin::plugin_startup` modified as part of this PR to log better error messages than `http_plugin startup fails, shutting down`.